Mathlib/Computability/ContextFreeGrammar.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2023 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak
+-/
+import Mathlib.Computability.Language
+
+/-!
+# Context-Free Grammar
+
+This file contains the definition of a Context-Free Grammar (CFG), which is a grammar that has a
+single nonterminal symbol on the left-hand side of each rule.
+Derivation by a grammar is inherently nondeterministic.
```

Comment:
Is this referring to an operation in a subsequent PR? If so, can you move this to a `TODO` section of the docstring?

If not, can you make sure "derivation" is included in the docstring of the thing you're referring to below?